Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -171,10 +171,14 @@ WarpX::AddSpaceChargeFieldLabFrame ()
     // Todo: use simpler finite difference form with beta=0
     std::array<Real, 3> beta = {0._rt};
 
+    // Check if an external Poisson solver is installed
+    auto callback_iter = warpx_callback_py_map.find("poissonsolver");
+    bool external_solver = callback_iter != warpx_callback_py_map.end();
+
     // Compute the potential phi, by solving the Poisson equation
-    if (warpx_py_poissonsolver) {
+    if ( external_solver ) {
```

Comment:
I can approve this as is, but I would suggest changing this to hide the implementation details of the map. Perhaps add a bool function `IsPythonCallBackInstalled`. Also, instead of `callback_iter->second();` I would just call `ExecutePythonCallback` - the time for the extra map lookup is trivial.

Reply:
Sounds good. I've made the suggested changes. Thanks for the review!